contracts/mixnet/src/contract.rs

```diff
@@ -207,43 +207,17 @@ pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> Result<QueryResponse, Cont
             gateway_identity,
             address,
         )?),
+        QueryMsg::GetTotalMixStake {} => to_binary(&queries::query_total_mix_stake(deps)),
+        QueryMsg::GetTotalGtStake {} => to_binary(&queries::query_total_gt_stake(deps)),
     };
 
     Ok(query_res?)
 }
-
 #[entry_point]
-pub fn migrate(deps: DepsMut, env: Env, _msg: MigrateMsg) -> Result<Response, ContractError> {
-    use crate::storage::{
-        all_gateway_delegations_read, all_mix_delegations_read, gateway_delegations,
-        mix_delegations, reverse_gateway_delegations, reverse_mix_delegations,
-    };
-    use mixnet_contract::{RawDelegationData, UnpackedDelegation};
-
-    let mix_bucket = all_mix_delegations_read::<Uint128>(deps.storage);
-    let delegations = Delegations::new(mix_bucket).collect::<Vec<UnpackedDelegation<Uint128>>>();
-    for delegation in delegations {
-        let owner = delegation.owner;
-        let node_identity = delegation.node_identity;
-        let amount = delegation.delegation_data;
-        let raw_delegation = RawDelegationData::new(amount, env.block.height);
-        mix_delegations(deps.storage, &node_identity).save(owner.as_bytes(), &raw_delegation)?;
-        reverse_mix_delegations(deps.storage, &owner).save(node_identity.as_bytes(), &())?;
-    }
-
-    let gateway_bucket = all_gateway_delegations_read::<Uint128>(deps.storage);
-    let delegations =
-        Delegations::new(gateway_bucket).collect::<Vec<UnpackedDelegation<Uint128>>>();
-    for delegation in delegations {
-        let owner = delegation.owner;
-        let node_identity = delegation.node_identity;
-        let amount = delegation.delegation_data;
-        let raw_delegation = RawDelegationData::new(amount, env.block.height);
-        gateway_delegations(deps.storage, &node_identity)
-            .save(owner.as_bytes(), &raw_delegation)?;
-        reverse_gateway_delegations(deps.storage, &owner).save(node_identity.as_bytes(), &())?;
-    }
+pub fn migrate(_deps: DepsMut, _env: Env, _msg: MigrateMsg) -> Result<Response, ContractError> {
+    todo!("Calculate initial total mix and gateway stake after initial deployment");
```

Comment:
Actually I never asked - is this  going to be in another pull request or something? Because technically after this PR is merged, any other contract changes will be blocked until the migration is implemented

Reply:
I think I'll merge this into the tokenomics branch first, then once we're ready to get that done we can preform the migration